### ISSUE TITLE
How variables defined in terragrunt is accessible in terraform code. 

### ISSUE THREAD

**Issue by irfanjs** [gruntwork-io/terragrunt#1270]
I have terragrunt config where have declared the variables using locals as below at root level. In the child module , have declared the child terragrunt config file called (terragrunt.hcl). parent terragrunt file has following code :

> locals {
>   location = "East US"
> }

child module terragrunt file has below code :

> 
> include {
>   path = find_in_parent_folders()
> }
> 
> locals {
>   myvars = read_terragrunt_config(find_in_parent_folders("terragrunt.hcl"))
>   location = local.myvars.locals.location
> }

now, trying to access location variable in the terraform code (main.tf) using following code :

>  location = "${var.location}"

but it throws error:

> Error: Reference to undeclared input variable
> 
>   on main.tf line 13, in resource "azurerm_resource_group" "example":
>   13:   location = "${var.location}"
> 

Not getting how i can access variables defined in the terragrunt file in the terraform code . please suggest

In short, i need this "location" variable to be defined in root terragrunt file and then refer it to the child module of terraform. can this be possible ? If yes, any example please .... Thanks 

NOTE: I do not want to define variables.tf file for common variables. common variables should be there in root terragrunt file and all child module should inherit that and be available in terraform code.

**Comment by yorinasub17** [comment]
The solution to this is to add a `generator` block to your root config that defines all the common variables. See [here](https://terragrunt.gruntwork.io/docs/features/keep-your-terraform-code-dry/#dry-common-terraform-code-with-terragrunt-generate-blocks) for the docs on how `generator` blocks work.

**Comment by yorinasub17** [comment]
Closing this as the original question has been answered. If there are any follow ups, please open a new ticket with updated context. Thanks!
